src/handlers/keyboard_handler.rs

```diff
@@ -0,0 +1,460 @@
+        Keysym::KP_1 => bevy::prelude::KeyCode::Numpad1,
+        Keysym::KP_2 => bevy::prelude::KeyCode::Numpad2,
+        Keysym::KP_3 => bevy::prelude::KeyCode::Numpad3,
+        Keysym::KP_4 => bevy::prelude::KeyCode::Numpad4,
+        Keysym::KP_5 => bevy::prelude::KeyCode::Numpad5,
+        Keysym::KP_6 => bevy::prelude::KeyCode::Numpad6,
+        Keysym::KP_7 => bevy::prelude::KeyCode::Numpad7,
+        Keysym::KP_8 => bevy::prelude::KeyCode::Numpad8,
+        Keysym::KP_9 => bevy::prelude::KeyCode::Numpad9,
+        Keysym::KP_Add => bevy::prelude::KeyCode::NumpadAdd,
+        Keysym::KP_Subtract => bevy::prelude::KeyCode::NumpadSubtract,
+        Keysym::KP_Multiply => bevy::prelude::KeyCode::NumpadMultiply,
+        Keysym::KP_Divide => bevy::prelude::KeyCode::NumpadDivide,
+        Keysym::KP_Decimal => bevy::prelude::KeyCode::NumpadDecimal,
+        Keysym::KP_Separator => bevy::prelude::KeyCode::NumpadComma, // Bevy uses NumpadComma for the separator
+        Keysym::KP_Equal => bevy::prelude::KeyCode::NumpadEqual,
+        // KP_Enter handled above with Return
+
+        // Punctuation & Symbols (Names can vary significantly)
+        Keysym::grave => bevy::prelude::KeyCode::Backquote, // `
+        Keysym::minus => bevy::prelude::KeyCode::Minus,     // -
+        Keysym::equal => bevy::prelude::KeyCode::Equal,     // =
+        Keysym::bracketleft => bevy::prelude::KeyCode::BracketLeft, // [
+        Keysym::bracketright => bevy::prelude::KeyCode::BracketRight, // ]
+        Keysym::backslash => bevy::prelude::KeyCode::Backslash, // \
+        Keysym::semicolon => bevy::prelude::KeyCode::Semicolon, // ;
+        Keysym::apostrophe => bevy::prelude::KeyCode::Quote, // ' (Bevy uses Quote)
+        Keysym::comma => bevy::prelude::KeyCode::Comma,     // ,
+        Keysym::period => bevy::prelude::KeyCode::Period,   // .
+        Keysym::slash => bevy::prelude::KeyCode::Slash,     // /
+
+        // Other Keys
+        Keysym::Print => bevy::prelude::KeyCode::PrintScreen,
+        Keysym::Pause => bevy::prelude::KeyCode::Pause,
+        Keysym::Menu => bevy::prelude::KeyCode::ContextMenu, // Bevy uses ContextMenu
+
+        // --- Fallback ---
+        // This arm catches any Keysym not explicitly handled above.
+        // Using todo!() will cause a panic, alerting you during development
+        // that a key mapping is missing. For production, consider returning Option<KeyCode>
+        // or logging an error instead of panicking.
+        _ => {
+            // e.g., eprintln!("Warning: Unhandled keysym: {:?}", keysym);
+            // If the function returned Option<KeyCode>, you'd return None here.
+            // Since it must return KeyCode, panic is the explicit way to fail.
+            todo!("Unhandled keysym: {:?}", keysym)
+            // Or potentially return a "default" like Escape, but this hides errors:
+            // KeyCode::Escape
+        }
+    }
+}
+
+impl KeyboardHandler for SmithayRunnerState {
+    fn enter(
```

Comment:
for every handler's function add inline document explaining what the function is and when is it triggered, that is an important inline documentation we dont want to miss.